### PR TITLE
feat(transformers): Add domain transformer for dataset

### PR DESCRIPTION
**Pull request by mohdsiddique**

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/transformer/dataset_domain.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    """Transformer that adds domains to datasets according to a callback function."""
+
+    ctx: PipelineContext
+    config: AddDatasetDomainConfig
+
+    def __init__(self, config: AddDatasetDomainConfig, ctx: PipelineContext):
+        super().__init__()
+        self.ctx = ctx
+        self.config = config
+        if self.config.semantics == Semantics.PATCH and self.ctx.graph is None:
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/transformer/dataset_domain.py:44]: Also resolve domains to fully qualified domain urns. 
See https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py#L513 for how this is done in other places.

**Review by shirshanka — DISMISSED** [review]
Added comment to resolve domain config to urns on init for all the transfomers.

File: metadata-ingestion/src/datahub/configuration/common.py [hunk 2]
```diff
@@ -11,6 +12,23 @@ class Config:
         extra = Extra.forbid
 
 
+class Semantics(Enum):
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/configuration/common.py:15]: maybe call this TransformerSemantics?

**Review by hsheth2 — COMMENTED** [review]
small nit but LGTM other than that

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**